src/logger.ts

```diff
@@ -1,7 +1,12 @@
+import type {WriteStream} from 'fs';
 import assert from 'assert';
 import chalk from 'chalk';
 
-const LEVEL = Object.freeze({
+export type LogFormatter = (data: LogEvent) => string;
+export interface LogContext { requestId?: string }
```

Comment:
The context is an arbitraty object, `requestId` is just the one our default formatter uses. It's a Node logging standard.

Reply:
You can keep `requestId?`, but it also needs to allow arbitrary pairs.